pennylane/devices/modifiers/simulator_tracking.py

```diff
@@ -0,0 +1,253 @@
+        if self.tracker.active:
+            batch = (circuits,) if isinstance(circuits, QuantumScript) else circuits
+            self.tracker.update(jvp_batches=1, jvps=len(batch))
+            self.tracker.record()
+        return untracked_compute_jvp(self, circuits, tangents, execution_config)
+
+    return compute_jvp
+
+
+def _track_execute_and_compute_jvp(untracked_execute_and_compute_jvp):
+    """Adds default tracking to a ``execute_and_compute_jvp`` method."""
+
+    @wraps(untracked_execute_and_compute_jvp)
+    def execute_and_compute_jvp(self, circuits, tangents, execution_config=DefaultExecutionConfig):
+        if self.tracker.active:
+            batch = (circuits,) if isinstance(circuits, QuantumScript) else circuits
+            for c in batch:
+                self.tracker.update(resources=c.specs["resources"])
+            self.tracker.update(execute_and_jvp_batches=1, executions=len(batch), jvps=len(batch))
+            self.tracker.record()
+
+        return untracked_execute_and_compute_jvp(self, circuits, tangents, execution_config)
+
+    return execute_and_compute_jvp
+
+
+def _track_compute_vjp(untracked_compute_vjp):
+    """Adds default tracking to a ``compute_vjp`` method."""
+
+    @wraps(untracked_compute_vjp)
+    def compute_vjp(self, circuits, cotangents, execution_config=DefaultExecutionConfig):
+        if self.tracker.active:
+            batch = (circuits,) if isinstance(circuits, QuantumScript) else circuits
+            self.tracker.update(vjp_batches=1, vjps=len(batch))
+            self.tracker.record()
+
+        return untracked_compute_vjp(self, circuits, cotangents, execution_config)
+
+    return compute_vjp
+
+
+def _track_execute_and_compute_vjp(untracked_execute_and_compute_vjp):
+    """Adds default trakcing to a ``execute_and_compute_vjp`` method."""
+
+    @wraps(untracked_execute_and_compute_vjp)
+    def execute_and_compute_vjp(
+        self, circuits, cotangents, execution_config=DefaultExecutionConfig
+    ):
+        if self.tracker.active:
+            batch = (circuits,) if isinstance(circuits, QuantumScript) else circuits
+            for c in batch:
+                self.tracker.update(resources=c.specs["resources"])
+            self.tracker.update(execute_and_vjp_batches=1, executions=len(batch), vjps=len(batch))
+            self.tracker.record()
+        return untracked_execute_and_compute_vjp(self, circuits, cotangents, execution_config)
+
+    return execute_and_compute_vjp
+
+
+# pylint: disable=protected-access
+def simulator_tracking(cls: type) -> type:
+    """Modifies all methods to add default simulator style tracking.
+
+    Args:
+        cls (type): a subclass of :class:`pennylane.devices.Device`
+
+    Returns
+        type: The inputted class that has now been modified to update the tracker upon function calls.
+
+    .. warning::
+
+        If used together with :func:`~.convert_single_circuit_to_batch`, that modifier must be applied first.
```

Comment:
why is this the case? can we just check if it's in `cls._applied_modifiers` here and raise an error if it won't work?

Reply:
Actually just realized it doesn't need to be first, so I'm removing the warning.